workshops/dodge/README.md

```diff
@@ -457,156 +463,235 @@ function mouseClicked() {
+function setup() {
+  ...
+  player = createSprite(width/2, height-25, 50, 50);
+  player.addImage(playerImage);
+  ...
+}
 ```
 
-And now our player has a bit more personality, and is more relatable. Try customizing the enemy as well!
+When we add an image to a sprite, the sprite's width and height is changed to match the image's. Because we're loading our sprite's image right after we create the sprite, we don't need to worry about the sprite's initial width and height. Let's just go ahead and set them to zero.
 
-## Part III. Code Cleanup
+```js
+function setup() {
+  ...
+  p̶l̶a̶y̶e̶r̶ ̶=̶ ̶c̶r̶e̶a̶t̶e̶S̶p̶r̶i̶t̶e̶(̶w̶i̶d̶t̶h̶/̶2̶,̶ ̶h̶e̶i̶g̶h̶t̶-̶2̶5̶,̶ ̶5̶0̶,̶ ̶5̶0̶)̶;̶
+  player = createSprite(width/2, height-25, 0, 0);
+  ...
+}
+```
 
-So we've used the various widths several times throughout our program, but it's almost impossible to tell by looking at the code, seeing as we never labeled them as such! What we have instead are these seemingly arbitrary numbers like "25" and "5" floating around. It's important to write code that's as clear and maintainable as possible, so that you (in the future) and others can quickly get up to speed and be able to make changes and improvements.
+We were previously setting the player sprite's initial y coordinate to half of the sprite's width away from the bottom of the screen. Now that we're using an image instead of a placeholder square, we can replace `25` with half of the player image's height.
+
+```js
+function setup() {
+  ...
+  p̶l̶a̶y̶e̶r̶ ̶=̶ ̶c̶r̶e̶a̶t̶e̶S̶p̶r̶i̶t̶e̶(̶w̶i̶d̶t̶h̶/̶2̶,̶ ̶h̶e̶i̶g̶h̶t̶-̶2̶5̶,̶ ̶0̶,̶ ̶0̶)̶;̶
+  player = createSprite(width/2, height-(playerImage.height/2), 0, 0);
+  ...
+}
+```
 
-We can get rid of these "magic numbers" by storing them as constants. Constants are variables that don't vary; i.e, they are set once and the value stays _constant_. First, we'll declare the existence of these constants, as we did our player and enemy variables:
+We're also going to want to make that same change in the `mouseClicked()` function.
 
 ```js
-var PLAYER_WIDTH;
-var PLAYER_HEIGHT;
-var ENEMY_WIDTH;
-var ENEMY_HEIGHT;
+function mouseClicked() {
+  ...
+  p̶l̶a̶y̶e̶r̶.̶p̶o̶s̶i̶t̶i̶o̶n̶.̶y̶ ̶=̶ ̶h̶e̶i̶g̶h̶t̶-̶2̶5̶;̶
+  player.position.y = height-(playerImage.height/2);
+  ...
+}
 ```
+##### Customizing the Enemy
 
-I've used all capitals and underscores, which is the convention for naming constants.
+Now do the same for the enemy:
 
-Next, we'll initialize them to values of our choice in `setup()`, and replace them in our creation of player and enemy, as well as in the `draw()` function.
+```js
+...
+var enemy;
+var enemyImage;
+
+function preload() {
+  ...
+  enemyImage = loadImage("https://surrogate.hackedu.us/i.imgur.com/OdL0XPt.png");
+}
+
+function setup() {
+  ...
+  enemy = createSprite(width/2, 0, 10, 30);
+  enemy.addImage(enemyImage);
+}
+```
+
+And remember to change references to the enemy's placeholder height with it's image's actual height.
```

Comment:
Ouch. Good catch.

On Fri, Apr 8, 2016 at 5:00 PM, JESSICA <notifications@github.com> wrote:

> In workshops/dodge/README.md
> <https://github.com/jkwok91/hackclub/pull/4#discussion_r59104328>:
>
> > +var enemy;
> > +var enemyImage;
> > +
> > +function preload() {
> > +  ...
> > +  enemyImage = loadImage("https://surrogate.hackedu.us/i.imgur.com/OdL0XPt.png");
> > +}
> > +
> > +function setup() {
> > +  ...
> > +  enemy = createSprite(width/2, 0, 10, 30);
> > +  enemy.addImage(enemyImage);
> > +}
> > +```
> > +
> > +And remember to change references to the enemy's placeholder height with it's image's actual height.
>
> its
>
> this is a reminder to myself for me to fix
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jkwok91/hackclub/pull/4/files/4c378ce7d8bec2b32e1d79ba9756c10b774a099f#r59104328>
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>